File: ios/Classes/TiktokEventsPlugin.swift
import Flutter
import TikTokBusinessSDK
import UIKit

public class TiktokEventsPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "tiktok_events", binaryMessenger: registrar.messenger())
    let instance = TiktokEventsPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "init":
      initSdk(call: call, result: result)
    case "setIdentity":
      setIdentity(call: call, result: result)
    case "logout":
      logout(call: call, result: result)
    default:     
      result(FlutterMethodNotImplemented)
    }
  }

  private func initSdk(call: FlutterMethodCall, result: @escaping FlutterResult) {
    guard let args = call.arguments as? [String: Any],
      let appId = args["appId"] as? String,
      let tiktokAppId = args["tiktokAppId"] as? String,
      let debug = args["debug"] as? Bool
    else {
      result(FlutterError(code: "INVALID_ARGUMENT", message: "Invalid arguments", details: nil))
      return
    }

    let config = TikTokConfig.init(appId: appId, tiktokAppId: tiktokAppId)
    if debug {
      config?.enableDebugMode()
    }

    TikTokBusiness.initializeSdk(config) { success, error in
      if !success {  // initialization failed
        result(
          FlutterError(
            code: "INITIALIZATION_FAILED", message: error?.localizedDescription, details: nil))
      } else {  // initialization successful
        result(nil)
      }
    }
  }

  private func setIdentity(call: FlutterMethodCall, result: @escaping FlutterResult) {
    guard let args = call.arguments as? [String: Any],
      let userId = args["userId"] as? String
    else {
      result(FlutterError(code: "INVALID_ARGUMENT", message: "Invalid arguments", details: nil))
      return
    }

    let userName = args["userName"] as? String
    let email = args["email"] as? String
    let phoneNumber = args["phoneNumber"] as? String

    TikTokBusiness.identify(withExternalID: userId, externalUserName: userName, phoneNumber: phoneNumber, email: email)
      
    result(nil)
  }

  private func logout(call: FlutterMethodCall, result: @escaping FlutterResult) {
    TikTokBusiness.logout()
    result(nil)
  }
}

File: android/src/main/kotlin/com/liangbx361/tiktok_events/TiktokEventsPlugin.kt
package com.liangbx361.tiktok_events

import android.content.Context
import com.tiktok.TikTokBusinessSdk
import com.tiktok.TikTokBusinessSdk.TTInitCallback
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

/**
 * Title:
 * Description:
 * Copyright © 2001-2024 17173. All rights reserved.
 *
 * @author liangbx
 * @version 2024/10/30
 */
class TiktokEventsPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var context: Context

    /// The MethodChannel that will the communication between Flutter and native Android
    ///
    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
    /// when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        context = binding.applicationContext
        channel = MethodChannel(binding.binaryMessenger, "tiktok_events")
        channel.setMethodCallHandler(this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "init" -> init(call, result)
            "setIdentify" -> setIdentify(call, result)
            "logout" -> logout(call, result)
            else -> result.notImplemented()
        }
    }

    private fun init(call: MethodCall, result: MethodChannel.Result) {
        val appId = call.argument<String>("appId")
        val tiktokAppId = call.argument<String>("tiktokAppId")
        val debug = call.argument<Boolean>("debug") ?: false

        val ttConfig = TikTokBusinessSdk.TTConfig(context)
            .setAppId(appId)
            .setTTAppId(tiktokAppId)
            .enableAutoIapTrack()

        if (debug) {
            ttConfig.openDebugMode()
        }

        TikTokBusinessSdk.initializeSdk(ttConfig, object : TTInitCallback {
            override fun success() {
                TikTokBusinessSdk.startTrack()
                result.success(null)
            }

            override fun fail(code: Int, message: String) {
                result.error(code.toString(), message, null)
            }
        })
    }

    private fun setIdentify(call: MethodCall, result: MethodChannel.Result) {
        val userId = call.argument<String>("userId")
        val userName = call.argument<String?>("userName")
        val email = call.argument<String?>("email")
        val phoneNumber = call.argument<String?>("phoneNumber")

        TikTokBusinessSdk.identify(userId, userName, email, phoneNumber)
        result.success(null)
    }

    private fun logout(call: MethodCall, result: MethodChannel.Result) {
        TikTokBusinessSdk.logout()
        result.success(null)
    }
}